File: src/main/java/mesi/orm/conn/DisposeableResultSet.java
package mesi.orm.conn;

import lombok.Getter;

import java.sql.ResultSet;
import java.sql.Statement;

/**
 * sql resultset wrapper
 * implements autocloseable
 * for java's try with resurce syntax
 */
public class DisposeableResultSet implements AutoCloseable {

    @Getter
    private final ResultSet resultSet;
    private final Statement statement;

    DisposeableResultSet(ResultSet resultSet, Statement statement) {
        this.resultSet = resultSet;
        this.statement = statement;
    }

    @Override
    public void close() throws Exception {
        resultSet.close();
        statement.close();
    }
}

File: orm-lib/src/main/kotlin/mesi/orm/conn/DisposeableResultSet.kt
package mesi.orm.conn

import java.sql.ResultSet
import java.sql.Statement

/**
 * disopseable resultset which should be used with 'use'
 */
internal class DisposeableResultSet(val resultSet: ResultSet, private val statement : Statement) : AutoCloseable {
    override fun close() {
        resultSet.close()
        statement.close()
    }
}